### DISCUSSION TITLE
Unraid Installation

### DISCUSSION THREAD

**Discussion by DrSpaldo** [BeardedTek-com/fEVR#44] [Q&A]
I would like to have a look at the project and was wondering if you had a suggested method of installation on a machine running Unraid?

**Comment by BeardedTek** [comment]
I have not used unRAID before. From what I gather, it is possible to install docker-compose  on unRAID even though it is not the recommended way to install things.

I'll experiment with unRAID this weekend and see what I can do!

**Comment by DrSpaldo** [comment]
That would be good. There are quite a few ways to tackle installing apps/software onto unraid. Here is a method that Sycotix from Ibracorp has used to install Tailscale onto unraid - https://www.youtube.com/watch?v=nzBQTJ2isOI

Also here are a few links so that you can add your own app into the unraid eco system, which means MUCH easier install for end users:

https://forums.unraid.net/topic/101424-how-to-publish-docker-templates-to-community-applications-on-unraid/
https://forums.unraid.net/topic/57181-docker-faq/#comment-566084

**Reply by BeardedTek** [reply]
Awesome!  Thank you for the input!

**Reply by markmghali** [reply]
my understanding it is a template for docker compose basically. I use unraid and would love to sue this in unraid! it would help expose your app to a wider audience as well.

Project looks great!

**Comment by DrSpaldo** [comment]
Anyone installed this on Unraid yet?

**Reply by s3ntech** [reply]
Just started looking into this myself, have you got this running yet? Getting this in the app store would definitely get this more exposure.  Hopefully someone with more skill than myself can eventually create a template.

**Reply by BeardedTek** [reply]
I just put the final touches on the flask rewrite release candidate today.  I might merge it as soon as tonight depending on how testing goes.  Once I do that I will get an unRAID template published and test it out as soon as I can.

**Reply by s3ntech** [reply]
That sounds great. Looking forward to it!

**Reply by BeardedTek** [reply]
You can add this repo to your unRAID to receive updates
https://github.com/BeardedTek-com/unraid-templates

Or download the template here:
https://raw.githubusercontent.com/BeardedTek-com/unraid-templates/main/beardedtek-com/fevr-v0.6-RC.xml

```
<?xml version="1.0"?>
<Container version="2">
  <Date>2022-04-18</Date>
  <Beta>True</Beta>
  <Name>fEVR</Name>
  <Repository>ghcr.io/beardedtek-com/fevr-flask:main</Repository>
  <Registry>https://github.com/BeardedTek-com/fEVR/pkgs/container/fevr</Registry>
  <Repository>beardedtek-com/fEVR-flask</Repository>
  <Project>https://fevr.video</Project>
  <Support>https://github.com/beardedtek-com/fEVR/discussions</Support>
  <Overview>fEVR works along side of frigate and home assistant to collect video and snapshots of objects detected using your existing camera systems.</Overview>
  <Category>HomeAutomation: Security:</Category>
  <Shell>bash</Shell>
  <Privileged>false</Privileged>
  <Network>bridge</Network>
  <WebUI>http://[IP]:[PORT:5090]/</WebUI>
  <Icon>https://fevr.video/img/logo.svg</Icon>
  <Description>fEVR works along side of frigate and home assistant to collect video and snapshots of objects detected using your existing camera systems.</Description>

  <Config Name="MQTT_CLIENT"
    Description="Runs fEVR mqtt client on the same docker container"
    Target="MQTT_CLIENT" Default="" Mode="" Type="Variable"
    Display="always" Required="true" Mask="false">true
  </Config>

  <Config Name="WebUI" 
    Description="Container Port: 5090"
    Target="5090" Default="" Mode="tcp" Type="Port"
    Display="always" Required="false" Mask="false"
  />

  <Config Name="PUID"
    Description="Container Variable: PUID"
    Target="PUID" Default="1000" Mode="" Type="Variable"
    Display="advanced-hide" Required="false" Mask="false">1000
  </Config>

  <Config Name="PGID"
    Description="Container Variable: PGID"
    Target="PGID" Default="1000" Mode="" Type="Variable"
    Display="advanced-hide" Required="false" Mask="false">1000
  </Config>

  <Config Name="Events"
    Target="/fevr/app/static/events"
    Default="/mnt/user/appdata/fevr/events" Mode="rw"
    Description=" Save events to a local folder" 
    Type="Path" Display="always" Required="false"
    Mask="false">/mnt/user/appdata/fevr/events
  </Config>

  <Config Name="FEVR_PORT"
    Description="Container Variable: FEVR_PORT"
    Target="FEVR_PORT" Default="5080" Mode="" Type="Variable"
    Display="advanced" Required="true" Mask="false">5090
  </Config>

  <Config Name="FLASK_ENV"
    Description="Set to 'development' for debug mode"
    Target="FLASK_ENV" Default="" Mode="" Type="Variable"
    Display="advanced" Required="false" Mask="false">
  </Config>

</Container>
```

**Reply by BeardedTek** [reply]
Still going through code and making things stable under one container.  This will make unRAID installation much simpler and everything under one workflow instead of 3 paths depending on install type.

Thanks for your patience.
